runtime/intern.c

```diff
@@ -262,6 +262,23 @@ static void intern_cleanup(struct caml_intern_state* s)
   intern_free_stack(s);
 }
 
+CAMLnoret static void intern_failwith2(const char * fun_name, const char * msg)
+{
+  char errmsg[100];
+  errmsg[sizeof(errmsg) - 1] = 0;
+  snprintf(errmsg, sizeof(errmsg) - 1, "%s: %s", fun_name, msg);
```

Comment:
```suggestion
  snprintf(errmsg, sizeof(errmsg), "%s: %s", fun_name, msg);
```

C99 `snprintf` always null-terminates, so (since truncation checking isn't necessary here), can this just be simplified to a one-liner now? Windows [snprintf is also C99 compliant now](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/snprintf-snprintf-snprintf-l-snwprintf-snwprintf-l?view=msvc-170) since MSVC 2015 and Windows 10, but I wouldn't be surprised to learn that's wrong in some corner case :-)

Reply:
You're right, this old code is weird and should not be reused.